### PR TITLE
PWG/EMCal: Simplification of cell calibrations and introduction of energy dependent time calib

### PR DESCRIPTION
**Pull request by FriederikeBock**
- removed multiple occurances of cell calibs to avoid future misconfigurations of the task
- added new functionality to calibrate the time per cell vs cell energy
- removed obsolete user task from PWGGA/EMCALtasks & PWGGA/GammaConv

### CODE REVIEW

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx [hunk 1]
```diff
@@ -57,6 +59,11 @@ Bool_t AliEmcalCorrectionCellTimeCalib::Initialize()
   if (!fRecoUtils)
     fRecoUtils  = new AliEMCALRecoUtils;
 
+  bool doCalibTimeEdep = kFALSE;
+  GetProperty("doCalibTimeEdep", doCalibTimeEdep);
+  fCalibrateTimeVsE = doCalibTimeEdep;
```
Comment (raymondEhlers) [inline PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx:64]: You can set `fCalibrateTimeVsE` directly with `GetProperty("doCalibTimeEdep", fCalibrateTimeVsE);` (unless I miss a negation somewhere or something)

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx [hunk 2]
```diff
@@ -146,6 +156,44 @@ Bool_t AliEmcalCorrectionCellTimeCalib::Run()
...
+/**
+ * Initialize the energy dependent time calibration.
+ */
+Int_t AliEmcalCorrectionCellTimeCalib::InitEDepTimeCalibration()
+{
+  
+  AliInfo("Initialising energy dependent time calibration map");
+  
+  std::unique_ptr<TFile> timeCalibFileVsE;
+  TSpline3* tiltCorr = NULL;
```
Comment (raymondEhlers) [inline PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx:169]: Better to use `nullptr`

File: PWG/EMCAL/EMCALbase/AliEMCALRecoUtils.h [hunk 3]
```diff
@@ -599,6 +620,11 @@ class AliEMCALRecoUtils : public AliEMCALRecoUtilsBase {
   TArrayL64  fGlobalEventID;                   ///< Global event ID
   Bool_t     fDoUseMergedBC;                   ///< flag for using one histo for all BCs
 
+  // energy dependent time clalibration
+  Bool_t     fTimeECorrection;            ///< Switch on or off the energy dependent time recalibration
+  TSpline3*  fEMCALTimeEShiftCorrection;  ///< Spline to correct energy dependent time shift for high gain cells
```
Comment (raymondEhlers) [inline PWG/EMCAL/EMCALbase/AliEMCALRecoUtils.h:625]: I didn't look closely at the RecoUtils, but I don't think you want this to stream, right? (ie. should be `//!<!`

**Review by raymondEhlers — COMMENTED** [review]
A few small comments

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx (hunk 2 above)
Comment (FriederikeBock) [reply PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx]: fixed

File: PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx (hunk 1 above)
Comment (FriederikeBock) [reply PWG/EMCAL/EMCALtasks/AliEmcalCorrectionCellTimeCalib.cxx]: fixed

File: PWG/EMCAL/EMCALbase/AliEMCALRecoUtils.h (hunk 3 above)
Comment (FriederikeBock) [reply PWG/EMCAL/EMCALbase/AliEMCALRecoUtils.h:625]: would prefer to leave this as is, as its done in the same manner for BC map and all other corrections

**Comment by raymondEhlers** [comment]
+1

**Comment by gconesab** [comment]
+1

**Pull request merged**